public/pages/PolicyDetails/containers/PolicyDetails/PolicyDetails.tsx

```diff
@@ -0,0 +1,243 @@
+ */
+
+import React, { Component } from "react";
+import {
+  EuiSpacer,
+  EuiTitle,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiButton,
+  EuiOverlayMask,
+  EuiButtonEmpty,
+  EuiModalFooter,
+  EuiModal,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiModalBody,
+  EuiCodeBlock,
+} from "@elastic/eui";
+import { RouteComponentProps } from "react-router-dom";
+import queryString from "query-string";
+import { PolicyService } from "../../../../services";
+import { BREADCRUMBS, ROUTES } from "../../../../utils/constants";
+import { getErrorMessage } from "../../../../utils/helpers";
+import PolicySettings from "../../components/PolicySettings/PolicySettings";
+import { ISMTemplate } from "../../../../../models/interfaces";
+import { CoreServicesContext } from "../../../../components/core_services";
+import DeleteModal from "../../components/DeleteModal/DeleteModal";
+import States from "../../../VisualCreatePolicy/components/States";
+
+interface PolicyDetailsProps extends RouteComponentProps {
+  policyService: PolicyService;
+}
+
+interface PolicyDetailsState {
+  policyId: string;
+  channelId: string;
+  primaryTerm: number;
+  lastUpdated: string;
+  description: string;
+  sequenceNumber: number;
+  schemaVersion: number;
+  ismTemplates: ISMTemplate[];
+
+  isJSONModalOpen: boolean;
+  policyJSON: any;
+  isDeleteModalVisible: boolean;
+}
+
+export default class PolicyDetails extends Component<PolicyDetailsProps,
+  PolicyDetailsState> {
+    static contextType = CoreServicesContext;
+    constructor(props: PolicyDetailsProps) {
+      super(props);
+
+      this.state = {
+        policyId: "",
+        channelId: "",
+        primaryTerm: -1,
+        lastUpdated: "",
+        description: "",
+        sequenceNumber: -1,
+        schemaVersion: -1,
+        ismTemplates: [],
+
+        isJSONModalOpen: false,
+        policyJSON: {policy:{states:[]}},
+        isDeleteModalVisible: false,
+      };
+    }
+
+    componentDidMount = async (): Promise<void> => {
+      this.context.chrome.setBreadcrumbs([BREADCRUMBS.INDEX_MANAGEMENT, BREADCRUMBS.INDEX_POLICIES]);
+      const { id } = queryString.parse(this.props.location.search);
+      if (typeof id === "string") {
+        this.context.chrome.setBreadcrumbs([BREADCRUMBS.INDEX_MANAGEMENT, BREADCRUMBS.INDEX_POLICIES, { text: id }]);
+        this.props.history.push(`${ROUTES.POLICY_DETAILS}?id=${id}`);
+        console.log("Getting policy!");
+        await this.getPolicy(id);
+        this.forceUpdate();
+      } else {
+        this.context.notifications.toasts.addDanger(`Invalid policy id: ${id}`);
+        this.props.history.push(ROUTES.INDEX_POLICIES);
+      }
+    };
+
+    getPolicy = async (policyId: string): Promise<void> => {
+      try {
+        const { policyService } = this.props;
+        const response = await policyService.getPolicy(policyId);
+
+        if (response.ok) {
+          console.log("Response: ", response.response);
+          this.setState({
```

Comment:
Why do we need to pull off all these fields from the policy (id, error notif, last updated time, description, schema version, ism templates) if you already have the policy itself stored in the state?